### PR TITLE
apply enhanced server-side tracing for httpgrpc.HTTP/Handle method

### PR DESCRIPTION
**Pull request by francoposa**
#### What this PR does

Applies the httpgrpc tracing middleware created in https://github.com/grafana/dskit/commit/0ef13fb65a5363eec6f14042d12554cf33cc8640.

This middleware increases the observability of http-over-grpc requests, which are all handled by the generic `httpgrpc.HTTP/Handle` method, by tagging the incoming `httpgrpc.HTTP/Handle` span with relevant opentracing http span tags, and by starting a child span mirroring the existing opentracing spans created for vanilla http requests.

This fixes the fact that all `httpgrpc.HTTP/Handle` tracing spans were identical, with no indication of which http request/endpoint type was being handled (e.g. /otlp/v1/metrics vs. /api/v1/push).

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,6 +78,8 @@
   * `-log.rate-limit-enabled`
   * `-log.rate-limit-logs-per-second`
   * `-log.rate-limit-logs-per-second-burst`
+* [ENHANCEMENT] Querier: cancel query requests to ingesters in a zone upon first error received from the zone, to reduce wasted effort spent computing results that won't be used #5764
+* [ENHANCEMENT] Apply dskit HTTPGRPCTracer for more detailed server-side tracing spans and tags on httpgrpc.HTTP/Handle requests #5782
```
Comment (charleskorn) [inline CHANGELOG.md:82]: Perhaps rephrase this to focus on the user impact? eg. "improve tracing of internal HTTP requests sent over httpgrpc"

**Review by charleskorn — APPROVED** [review]
LGTM modulo changelog suggestion

File: CHANGELOG.md [hunk 2]
```diff
@@ -78,6 +78,8 @@
   * `-log.rate-limit-enabled`
   * `-log.rate-limit-logs-per-second`
   * `-log.rate-limit-logs-per-second-burst`
+* [ENHANCEMENT] Querier: cancel query requests to ingesters in a zone upon first error received from the zone, to reduce wasted effort spent computing results that won't be used #5764
```
Comment (jhesketh) [inline CHANGELOG.md:81]: I'm confused by this line. Is this left over from something? The PR referenced is for the above changelog line.

File: CHANGELOG.md (hunk 1 above)
Comment (francoposa) [reply CHANGELOG.md]: updated in latest commit
